test/server/http_filter_integration_test_base.cc

```diff
@@ -0,0 +1,72 @@
+#include "test/server/http_filter_integration_test_base.h"
+
+#include "gtest/gtest.h"
+
+namespace Nighthawk {
+
+HttpFilterIntegrationTestBase::HttpFilterIntegrationTestBase(
+    Envoy::Network::Address::IpVersion ip_version)
+    : HttpIntegrationTest(Envoy::Http::CodecClient::Type::HTTP1, ip_version),
+      request_headers_({{":method", "GET"}, {":path", "/"}, {":authority", "host"}}) {}
+
+void HttpFilterIntegrationTestBase::setup(const std::string& config) {
+  config_helper_.addFilter(config);
+  HttpIntegrationTest::initialize();
+}
+
+Envoy::IntegrationStreamDecoderPtr HttpFilterIntegrationTestBase::getResponseFromUpstream() {
+  return getResponse(request_headers_, true);
+}
+
+Envoy::IntegrationStreamDecoderPtr
+HttpFilterIntegrationTestBase::getResponseFromUpstream(absl::string_view request_level_config) {
+  return getResponse(request_level_config, true);
+}
+
+Envoy::IntegrationStreamDecoderPtr
+HttpFilterIntegrationTestBase::getResponseFromExtension(absl::string_view request_level_config) {
+  return getResponse(request_level_config, false);
+}
+
+Envoy::IntegrationStreamDecoderPtr HttpFilterIntegrationTestBase::getResponseFromUpstream(
+    const Envoy::Http::TestRequestHeaderMapImpl& request_headers) {
+  return getResponse(request_headers, true);
+}
+
+Envoy::IntegrationStreamDecoderPtr HttpFilterIntegrationTestBase::getResponseFromExtension(
+    const Envoy::Http::TestRequestHeaderMapImpl& request_headers) {
+  return getResponse(request_headers, false);
+}
+
+Envoy::IntegrationStreamDecoderPtr
+HttpFilterIntegrationTestBase::getResponse(absl::string_view request_level_config,
+                                           bool setup_for_upstream_request) {
+  const Envoy::Http::LowerCaseString key("x-nighthawk-test-server-config");
```

Comment:
It would be good to mention in all doc comments that `request_level_config` is sent as the `x-nighthawk-test-server-config` header -- I didn't realize how it worked.